### PR TITLE
Add celltype report to downloads

### LINKED ISSUE #496: Cell Type Annotation
### Context

This issue is to collect issues around adding Cell Type Annotation support on the ScPCA Portal.

This epic is "blocked" by the Anndata Support Epic only because we want to start this work after that is deployed to production.

Test data is available:
https://github.com/AlexsLemonade/ScPCA-admin/pull/717

### PR DESCRIPTION
**Pull request by arkid15r**
## Issue Number

Resolves #496 

## Purpose/Implementation Notes

Adds celltype report file for single-cell/multiplexed modalities when project has no cell lines.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/scpca_portal/models/computed_file.py [hunk 1]
```diff
@@ -246,8 +246,17 @@ def get_sample_multiplexed_file(
...
             library_id = library["scpca_library_id"]
-            for file_suffix in ("_filtered.rds", "_processed.rds", "_qc.html", "_unfiltered.rds"):
-                file_name = f"{library_id}{file_suffix}"
+            file_suffixes = [
+                "filtered.rds",
+                "processed.rds",
+                "qc.html",
+                "unfiltered.rds",
+            ]
+            if not sample.project.includes_cell_lines:
```
Comment (davidsmejia) [inline api/scpca_portal/models/computed_file.py:255]: This should be based off of the sample/library and not the project.

`project.includes_cell_lines` just means that at least one sample/library where `sample.is_cell_line` is true.

The value that we want to base this off of can be found here:
https://github.com/AlexsLemonade/ScPCA-admin/blob/main/scpca-test-data/sample-metadata.tsv

File: api/scpca_portal/models/computed_file.py [hunk 2]
```diff
@@ -280,20 +289,23 @@ def get_sample_single_cell_file(cls, sample, libraries, workflow_versions, file_
...
             file_name = sample.output_single_cell_computed_file_name
-            common_file_suffixes = (
+            common_file_suffixes = [
                 "filtered.rds",
                 "processed.rds",
                 "qc.html",
                 "unfiltered.rds",
-            )
+            ]
+        if not sample.project.includes_cell_lines:
```
Comment (davidsmejia) [inline api/scpca_portal/models/computed_file.py:306]: same comment as above

**Review by davidsmejia — COMMENTED** [review]
Looking good, just need to make sure we check off of the sample and not the project.

**Review by arkid15r — COMMENTED** [review]
I've changed the processing logic based on your comment. PTAL

File: api/scpca_portal/models/computed_file.py (hunk 2 above)
Comment (arkid15r) [reply api/scpca_portal/models/computed_file.py]: Done.

File: api/scpca_portal/models/computed_file.py (hunk 1 above)
Comment (arkid15r) [reply api/scpca_portal/models/computed_file.py]: Done.

**Pull request merged**